fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/MetricUnionAvailability.java

```diff
@@ -111,8 +112,7 @@ public SimplifiedIntervalList getAvailableIntervals(PhysicalDataSourceConstraint
                 .flatMap(Set::stream)
                 .collect(Collectors.toSet());
 
-        // If the table is configured with a column that is not supported by the underlying data sources
-        if (!constraint.getMetricNames().stream().allMatch(dataSourceMetricNames::contains)) {
+        if (hasUnconfiguredMetric(constraint, dataSourceMetricNames)) {
```

Comment:
`!constraint.getMetricNames().stream().allMatch(dataSourceMetricNames::contains)` now becomes a helper method in `getAvailabilitiesToMetricNames()` below, because `MetricLeftUnionAvailability` also needs this check